tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -1495,102 +1491,43 @@ class SpirvFunctionCall : public SpirvInstruction {
-class SpirvNonUniformBinaryOp : public SpirvGroupNonUniformOp {
-public:
-  SpirvNonUniformBinaryOp(spv::Op opcode, QualType resultType,
-                          SourceLocation loc, spv::Scope scope,
-                          SpirvInstruction *arg1, SpirvInstruction *arg2);
-
-  DEFINE_RELEASE_MEMORY_FOR_CLASS(SpirvNonUniformBinaryOp)
+  DEFINE_RELEASE_MEMORY_FOR_CLASS(SpirvGroupNonUniformOp)
 
   // For LLVM-style RTTI
   static bool classof(const SpirvInstruction *inst) {
-    return inst->getKind() == IK_GroupNonUniformBinaryOp;
+    return inst->getKind() == IK_GroupNonUniformOp;
   }
 
   bool invokeVisitor(Visitor *v) override;
 
-  SpirvInstruction *getArg1() const { return arg1; }
-  SpirvInstruction *getArg2() const { return arg2; }
-  void replaceOperand(
-      llvm::function_ref<SpirvInstruction *(SpirvInstruction *)> remapOp,
-      bool inEntryFunctionWrapper) override {
-    arg1 = remapOp(arg1);
-    arg2 = remapOp(arg2);
-  }
-
-private:
-  SpirvInstruction *arg1;
-  SpirvInstruction *arg2;
-};
-
-/// \brief OpGroupNonUniformElect instruction. This is currently the only
-/// non-uniform instruction that takes no other arguments.
-class SpirvNonUniformElect : public SpirvGroupNonUniformOp {
-public:
-  SpirvNonUniformElect(QualType resultType, SourceLocation loc,
-                       spv::Scope scope);
-
-  DEFINE_RELEASE_MEMORY_FOR_CLASS(SpirvNonUniformElect)
-
-  // For LLVM-style RTTI
-  static bool classof(const SpirvInstruction *inst) {
-    return inst->getKind() == IK_GroupNonUniformElect;
-  }
-
-  bool invokeVisitor(Visitor *v) override;
-};
-
-/// \brief OpGroupNonUniform* unary instructions.
-class SpirvNonUniformUnaryOp : public SpirvGroupNonUniformOp {
-public:
-  SpirvNonUniformUnaryOp(spv::Op opcode, QualType resultType,
-                         SourceLocation loc, spv::Scope scope,
-                         llvm::Optional<spv::GroupOperation> group,
-                         SpirvInstruction *arg);
-
-  DEFINE_RELEASE_MEMORY_FOR_CLASS(SpirvNonUniformUnaryOp)
-
-  // For LLVM-style RTTI
-  static bool classof(const SpirvInstruction *inst) {
-    return inst->getKind() == IK_GroupNonUniformUnaryOp;
-  }
+  spv::Scope getExecutionScope() const { return execScope; }
 
-  bool invokeVisitor(Visitor *v) override;
+  llvm::ArrayRef<SpirvInstruction *> getOperands() const { return operands; }
 
-  SpirvInstruction *getArg() const { return arg; }
   bool hasGroupOp() const { return groupOp.hasValue(); }
   spv::GroupOperation getGroupOp() const { return groupOp.getValue(); }
+
   void replaceOperand(
       llvm::function_ref<SpirvInstruction *(SpirvInstruction *)> remapOp,
       bool inEntryFunctionWrapper) override {
-    arg = remapOp(arg);
-    if (inEntryFunctionWrapper)
-      setAstResultType(arg->getAstResultType());
+    for (auto operand : getOperands()) {
+      operand = remapOp(operand);
+      if (inEntryFunctionWrapper)
+        setAstResultType(operand->getAstResultType());
```

Comment:
This will `setAstResultType` for each operand, where formerly it was only doing so for the one arg (for `SpirvNonUniformUnaryOp`). Is this okay?

Reply:
Update: This is the bit I was referring to as the "PervertexInputVisitor" code (the only place this function is called from), because yeah, this isn't a great answer.

Reply:
It's fine, I don't want to hold up the change. Just wanted to call your attention to this, in case it's a problem.